docs/features/deconstruction.md

```diff
@@ -0,0 +1,180 @@
+```
+evaluate side-effect on the left-hand-side variables
+evaluate Deconstruct passing the references directly in
+```
+
+Note that the feature is built around the `Deconstruct` mechanism for deconstructing types.
+`ValueTuple` and `System.Tuple` will rely on that same mechanism, except that the compiler may need to synthesize the proper `Deconstruct` methods.
+
+
+**Work items, open issues and assumptions to confirm with LDM:**
+
+- I assume this should work even if `System.ValueTuple` is not present.
+- How is the Deconstruct method resolved?
+    - I assumed there can be no ambiguity. Only one `Deconstruct` is allowed (in nesting cases we have no type to guide the resolution process).
+    - But we may allow a little bit of ambiguity and preferring an instance over extension method.
+- Do the names matter? `int x, y; (a: x, b: y) = M();`
+- Can we deconstruct into a single out variable? I assume no.
+- I assume no compound assignment `(x, y) += M();`
+- [ ] Provide more details on the semantic of deconstruction-assignment, both static (The LHS of the an assignment-expression used be a L-value, but now it can be L-value -- which uses existing rules -- or tuple_literal. The new rules for tuple_literal on the LHS...) and dynamic.
+- [ ] Discuss with Aleksey about "Deconstruct and flow analysis for nullable ref type"
+- [ ] Validate any target typing or type inference scenarios.
+- The deconstruction-assignment is treated separately from deconstruction-declaration, which means it doesn't allow combinations such as `int x; (x, int y) = M();`.
+
+###Deconstruction-declaration (deconstruction into new variables):
+
+```ANTLR
+declaration_statement
+    : local_variable_declaration ';'
+    | local_constant_declaration ';'
+    | local_variable_combo_declaration ';'  // new
+    ;
+
+local_variable_combo_declaration
+    : local_variable_combo_declaration_lhs '=' expression
+
+local_variable_combo_declaration_lhs
+    : 'var' '(' identifier_list ')'
+    | '(' local_variable_list ')'
+    ;
+
+identifier_list
+    : identifier ',' identifier
+    | identifier_list ',' identifier
+    ;
+
+local_variable_list
+    : local_variable_type identifier ',' local_variable_type identifier
+    | local_variable_list ',' local_variable_type identifier
+    ;
+
+foreach_statement
+    : 'foreach' '(' local_variable_type identifier 'in' expression ')' embedded_statement
+    | 'foreach' '(' local_variable_combo_declaration_lhs 'in' expression ')' embedded_statement // new
+    ;
+
+for_initializer
+    : local_variable_declaration
+    | local_variable_combo_declaration // new
+    | statement_expression_list
+    ;
+
+let_clause
+    : 'let' identifier '=' expression
+    | 'let' '(' identifier_list ')' '=' expression // new
+    ;
+
+from_clause // not sure
+    : 'from' type? identifier 'in' expression
+    ;
+
+join_clause // not sure
+    : 'join' type? identifier 'in' expression 'on' expression 'equals' expression
```

Comment:
The list above resulted from my fishing for syntax nodes that look like they declare variables and so would be candidates for deconstruction-declaration.
For those marked "not sure", I wasn't sure whether they are good candidates and what the updated syntax would be.